gensim/models/phrases.py

```diff
@@ -106,17 +206,23 @@ def __init__(self, sentences=None, min_count=5, threshold=10.0,
         `delimiter` is the glue character used to join collocation tokens, and
         should be a byte string (e.g. b'_').
 
+        `average_sentence_size`: expected number of tokens in average sentence.
+            max_vocab_size / average_sentence_size sentences is used to build python
+            dictionary per thread.
+            Lowering `average_sentence_size` will increase memory demands.
         """
         if min_count <= 0:
-            raise ValueError("min_count should be at least 1")
-
+            raise ValueError("min_count should be 1")
+        if min_count > 1:
+            logger.warning("min_count should be 1")
```

Comment:
In my opinion, when the underlying behavior changes, it can be more safe/honest to break the API (give a "no such parameter min_count" error) than maintain a superficial compatibility that's no longer having the original effect (silently altering the API) or indeed fails (with a thrown error) in most situations where the old values would have been supported. 

Perhaps here it's better for both implementations to co-exist, at least for a while, as different classes with slightly-different options? That may also make it easier to compare their performance, and allow any project that prefers the old precision, or needs reproducibility of prior runs, to stay with the original implementation unless/until they want the benefits of the new.